pkg/apply/task/apply_task.go

```diff
@@ -116,6 +127,22 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 					object.UnstructuredToObjMeta(obj), event.Unchanged, applyerror.NewUnknownTypeError(err))
 				continue
 			}
+
+			clusterObj, err := getClusterObj(dynamic, info)
```

Comment:
It is a bit unfortunate that we need this, but I'm not sure if there is a way to avoid it for now. The implementation of `ApplyOptions.Run` also does a get, so we end up fetching every object twice.

Reply:
We could avoid it by not using the kubectl `ApplyOptions`,  but it would require coping lots of kubectl code. Let's keep this implementation. If we see any user report performance issue, we can improve it then.